NSS-Enthusiastic-Camels/BangazonAPI
# Pg employee

## Link to Ticket
closes # [Link](https://github.com/NSS-Enthusiastic-Camels/BangazonAPI/issues/7)

## Description of Proposed Changes

- Create model for Employee
- Created mvc for Employee


## Steps to Test
1.
git fetch --all
git checkout dsSetUp

2.
Fill in a few data in sql.  below.

insert into Employee
(FirstName,LastName,DepartmentId,IsSuperVisor)
VALUES ('Mark','Hale',1,1);

3.
Test Get all:
in postman get with URL

http://localhost:5000/api/Employee

4.
Test Get single:
in postman get with URL

http://localhost:5000/api/Employee/2

5.

Test Post:
in postman post with URL

http://localhost:5000/api/Employee

in body paste

  {
        "firstName": "Mark",
        "lastName": "Hale",
        "isSuperVisor": 1,
        "departmentId": 2,
        }

Get Employees again to see outcome.

6.
Test Put:
in postman put with URL

http://localhost:5000/api/Employee/{id}
in body paste

    {
        "firstName": "Mark",
        "lastName": "Hale",
        "isSuperVisor": 1,
        "departmentId": 1
        }

Get Employeet to see outcome.


## Impacted Areas in Application

List general components of the application that this PR will affect:

* EmployeeModel and EmployeeController


## Mentions @username

@alexfont321
@stjepad
@markhale94


## Definition of Done

1. The project must be fully documented. This includes the following:
    1. Complete README that documents the steps to install the code, how to install any dependencies, 
       or system configuration needed.
    2. Every class must be documented with purpose, author, and methods.
    3. Every method must be documented with purpose and argument list - which itself must contain a 
       short purpose for each argument.
2. The project must be able to run fully, and without errors, on each teammate's system.
3. Fulfills every requirement.
    Every line of code has been peer reviewed.
4. For projects that require unit testing, core functionality must be identified and have at least one test 
    for each.
